utopia-api/src/primitives/common.tsx

```diff
@@ -1,254 +1,11 @@
-      // WARNING: groups can only support T-L-W-H children
-      const childFrames = computeGroupChildren(children)
-      const selfFrame = calculateGroupBounds(childFrames)
-      const shiftedChildFrames = shiftGroupChildren(selfFrame, childFrames)
-      const childPositionStyles = shiftedChildFrames.map((childFrame) => {
-        if (childFrame == null) {
-          return {}
-        } else {
-          return convertPinsToAbsoluteStyleProps(childFrame)
-        }
-      })
-      return childPositionStyles
-    case LayoutSystem.PinSystem:
-    case undefined:
-      return []
-    default:
-      const _exhaustiveCheck: never = layout.layoutSystem
-      throw new Error(`Unknown layout type ${JSON.stringify(layout.layoutSystem)}`)
-  }
-}
-
-function calculateContainerProps(
-  parentLayoutSystem: LayoutSystem | undefined,
-  props: React.PropsWithChildren<UtopiaComponentProps>,
-  children: React.ReactNode,
-): CSSFrame {
-  const layout = defaultIfNull({} as LayoutProps, props.layout)
-  switch (layout.layoutSystem) {
-    case LayoutSystem.Group:
-      const childFrames = computeGroupChildren(children)
-      const selfFrame = calculateGroupBounds(childFrames)
-      return {
-        position: 'absolute',
-        ...selfFrame,
-      }
-    case LayoutSystem.PinSystem:
-      return {
-        position: parentLayoutSystem === LayoutSystem.PinSystem ? 'absolute' : 'relative',
-      }
-    case undefined:
-      return flexParentPropsToStyle(layout)
-    default:
-      const _exhaustiveCheck: never = layout.layoutSystem
-      throw new Error(`Unknown layout type ${JSON.stringify(layout.layoutSystem)}`)
-  }
-}
-
-export function calculateOwnStyleProp(
-  props: React.PropsWithChildren<UtopiaComponentProps>,
-  children: React.ReactNode,
-): CSSFrame {
-  if (props.layout == null) {
-    return {}
-  } else {
-    const layout = { ...props.layout, ...props.style } as LayoutProps
-    switch (layout.layoutSystem) {
-      case LayoutSystem.Group:
-        // TODO more group stuff
-        const childFrames = computeGroupChildren(children)
-        const selfFrame = calculateGroupBounds(childFrames)
-        return {
-          position: 'absolute',
-          ...selfFrame,
-        }
-      case LayoutSystem.PinSystem:
-        return {
-          ...flexElementPropsToStyle(layout, {}, 0, 1), // TODO instead of flexElementPropsToStyle, make a function that moves only the non-magic props to style
-          ...flexParentPropsToStyle(layout),
-          ...convertPinsToStyleProps(layout as any),
-          position: 'relative',
-        }
-      default:
-        return {
-          ...flexElementPropsToStyle(layout, {}, 0, 1), // TODO instead of flexElementPropsToStyle, make a function that moves only the non-magic props to style
-          ...flexParentPropsToStyle(layout),
-          ...convertPinsToStyleProps(layout as any),
-        }
-    }
-  }
-}
-
 export function addEventHandlersToDivProps(
```

Comment:
wow View does _this_? I think at this point this is more likely to just catch out someone rather than help them?